catroid/src/main/java/org/catrobat/catroid/content/actions/ChangeXByNAction.java

```diff
@@ -40,6 +41,11 @@ protected void update(float percent) {
 		try {
 			Float newX = xMovement == null ? Float.valueOf(0f) : xMovement.interpretFloat(sprite);
 			sprite.look.changeXInUserInterfaceDimensionUnit(newX);
+
+			if (sprite.penConfiguration.penDown) {
+				sprite.penConfiguration.pointsToDraw.add(new PointF(sprite.look.getXInUserInterfaceDimensionUnit(),
+						sprite.look.getYInUserInterfaceDimensionUnit()));
+			}
```

Comment:
This approach won't work with the physics engine because position changes aren't triggered by actions.
Also actions are easily missed (e.g. IfOnEdgeBounceAction) and developers will forget to add it for new bricks/features.